benchmarks/Csharp/Benchmarks/NBody.cs

```diff
@@ -0,0 +1,195 @@
+{
+    private readonly Body[] bodies;
+
+    public NBodySystem()
+    {
+        bodies = CreateBodies();
+    }
+
+    public static Body[] CreateBodies()
+    {
+        Body[] bodies = new[] {Body.Sun,
+                            Body.Jupiter,
+                            Body.Saturn,
+                            Body.Uranus,
+                            Body.Neptune};
+
+        var px = 0.0;
+        var py = 0.0;
+        var pz = 0.0;
+
+        foreach (var b in bodies)
+        {
+            px += b.Vx * b.Mass;
+            py += b.Vy * b.Mass;
+            pz += b.Vz * b.Mass;
+        }
+
+        bodies[0].OffsetMomentum(px, py, pz);
+
+        return bodies;
+    }
+
+    public void Advance(double dt)
+    {
+        for (var i = 0; i < bodies.Length; i++)
+        {
+            for (var j = i + 1; j < bodies.Length; j++)
+                CalculateVelocities(bodies[i], bodies[j]);
+        }
+
+        foreach (var body in bodies)
+        {
+            body.X += dt * body.Vx;
+            body.Y += dt * body.Vy;
+            body.Z += dt * body.Vz;
+        }
+
+        void CalculateVelocities(Body a, Body b)
+        {
+            var dx = a.X - b.X;
+            var dy = a.Y - b.Y;
+            var dz = a.Z - b.Z;
+            
+            var dSquared = (dx * dx) + (dy * dy) + (dz * dz);
+            var distance = Math.Sqrt(dSquared);
+            var mag = dt / (dSquared * distance);
+
+            a.Vx -= dx * b.Mass * mag;
+            a.Vy -= dy * b.Mass * mag;
+            a.Vz -= dz * b.Mass * mag;
+
+            b.Vx += dx * a.Mass * mag;
+            b.Vy += dy * a.Mass * mag;
+            b.Vz += dz * a.Mass * mag;
+        }
+    }
+
+    public double Energy()
+    {
+        var e = 0.0;
+
+        for (var i = 0; i < bodies.Length; i++)
+        {
+            var iBody = bodies[i];
+            e += 0.5 * iBody.Mass
+                * ((iBody.Vx * iBody.Vx) +
+                   (iBody.Vy * iBody.Vy) +
+                   (iBody.Vz * iBody.Vz));
+
+            for (var j = i + 1; j < bodies.Length; j++)
+            {
+                var jBody = bodies[j];
+                var dx = iBody.X - jBody.X;
+                var dy = iBody.Y - jBody.Y;
+                var dz = iBody.Z - jBody.Z;
+
+                var distance = Math.Sqrt((dx * dx) + (dy * dy) + (dz * dz));
+                e -= iBody.Mass * jBody.Mass / distance;
+            }
+        }
+        return e;
+    }
+}
+
+public sealed class Body
+{
+    private const double PI = 3.141592653589793;
+    private const double SOLAR_MASS = 4 * PI * PI;
+    private const double DAYS_PER_YER = 365.24;
+
+    public Body(double x, double y, double z, double vx, double vy, double vz, double mass)
+    {
+        X = x;
+        Y = y;
+        Z = z;
+        Vx = vx * DAYS_PER_YER;
+        Vy = vy * DAYS_PER_YER;
+        Vz = vz * DAYS_PER_YER;
+        Mass = mass * SOLAR_MASS;
+    }
+
+    public double X { get; set; }
```

Comment:
It's not sufficient to define this as a public field?

Reply:
That would be perfectly sufficient, although it would also be against MS ""

Java also used the property pattern i.e.:
```
  private double x;
  public double getX() { return x; }
  public void setX(final double x) { this.x = x; }
```
C# property is the equivalent for Java field with get & set methods

Reply:
I'll happily defer to your or the linter's judgement of what's idiomatic.

My thought was, especially when compared to Java: in C# you don't need to change the client code when switching from `public double X` to `public double X {...}`, so, a field might be fine. But indeed, hopefully the compiler will eliminate the difference for trivial ones already anyway. So, using a full property is also a good test to make sure the compiler actually does it.

Reply:
Changing a field to a property in C# is not breaking if (and only if) you can re-compile all the consumers. (or it is source-compatible but binary-breaking)
As far as I know, the binary breaking change is the reason MS recommends properties.

Reply:
Ah, right, very good reason to go with properties then :)